source/MainWindow.cpp

```diff
@@ -1424,30 +1463,130 @@ MainWindow::_BuildLine() // ask all the views what they hold, reset the command
+		value = fAudioCodecs[option_index].Option;
+		_SetParameter("-acodec", value);
+
 		if (option_index != 0) {
-			fCommand << " -b:a " << std::atoi(fAudioBitsPopup->FindMarked()->Label()) << "k";
-			fCommand << " -ar " << std::atoi(fSampleratePopup->FindMarked()->Label());
-			fCommand << " -ac " << fChannelCount->Value();
-			fCommand << " -strict -2 "; // enable 'experimental codecs' needed for dca (DTS)
+			value = fAudioBitsPopup->FindMarked()->Label();
+			value << "k";
+			_SetParameter("-b:a", value);
+			value = fSampleratePopup->FindMarked()->Label();
+			_SetParameter("-ar", value);
+			value = "";
+			value << fChannelCount->Value();
+			_SetParameter("-ac", value);
+			_SetParameter("-strict", "-2"); // enable 'experimental codecs' needed for dca (DTS)
 		}
-	} else
-		fCommand << (" -an");
+		else {
+			_RemoveParameter("-b:a");
+			_RemoveParameter("-ar");
+			_RemoveParameter("-ac");
+		}
+	}
+	else {
+		_RemoveParameter("-acodec");
+		_RemoveParameter("-b:a");
+		_RemoveParameter("-ar");
+		_RemoveParameter("-ac");
+		_SetParameter("-an", "");
+	}
+
+	//logging and output formatting
+	_SetParameter("-loglevel", "error");
+	_SetParameter("-stats","");
 
-	fCommand << " \"" << output_filename << "\"";
-	fCommand << " -loglevel error -stats";
-	fCommandlineTextControl->SetText(fCommand.String());
+	// output file
+	BString output_filename = fOutputTextControl->Text();
+	output_filename.Trim();
+	output_filename.Prepend("\"");
+	output_filename.Append("\"");
+
+	// can´t use _SetParameter() here because the output file is specified without an option specifier
+	// for now we just assume that the output file is at the end of the command line and in double quotes
+	// this should definitely be improved in the future
+	int32 last_idx = fCommandLineTokens.CountStrings() - 1;
+	BString last_item = fCommandLineTokens.StringAt(last_idx);
+	if (last_item.StartsWith("\"") and last_item.EndsWith("\""))
+		fCommandLineTokens.Replace(last_idx, output_filename);
+	else
+		fCommandLineTokens.Add(output_filename);
+
+	// assemble the commandline from the token list and put it in the textcontrol
+	fCommand.SetTo("");
+
+	for (int32 i=0; i<fCommandLineTokens.CountStrings(); ++i)
+		fCommand << fCommandLineTokens.StringAt(i) << " ";
+
+	fCommandlineTextControl->SetText(fCommand);
+}
+
+
+void
+MainWindow::_SetParameter(const BString& name, const BString& value)
+{
+	if (!fCommandLineTokens.HasString(name)) {
+		BString last_token = fCommandLineTokens.Last();
+		if (last_token.StartsWith("\"") and last_token.EndsWith("\""))
+			fCommandLineTokens.Add(name, fCommandLineTokens.CountStrings() - 1);
+		else
+			fCommandLineTokens.Add(name);
+	}
+
+	if (!value.IsEmpty()) {
+		int32 param_index = fCommandLineTokens.IndexOf(name);
+		BString next_token = fCommandLineTokens.StringAt(param_index + 1);
+
+		if ((next_token.StartsWith("-") and !_IsDigit(next_token.ByteAt(1))))  // no parameter value
+			fCommandLineTokens.Add(value, param_index+1);
+		else
```

Comment:
I think we do need curly brackets here, because the else-block stretches over several lines.